test/e2e/framework/job/wait.go

```diff
@@ -68,6 +68,17 @@ func WaitForJobComplete(ctx context.Context, c clientset.Interface, ns, jobName
 	})
 }
 
+// WaitForJobFailed uses c to wait for the Job jobName in namespace ns to fail
+func WaitForJobReady(ctx context.Context, c clientset.Interface, ns, jobName string, ready int32) error {
+	return wait.PollWithContext(ctx, framework.Poll, JobTimeout, func(ctx context.Context) (bool, error) {
+		curr, err := c.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return *curr.Status.Ready == ready, nil
```

Comment:
I think we should also make sure it is not `nil` to avoid potential race conditions. 
Maybe pass `*int32` to the function and verify `ptr.Equal(ready, job.Status.Ready)`

Reply:
updated https://github.com/kubernetes/kubernetes/commit/bf5540847e8b8bce8b4846e9441303df790d5738